services/_internal/maven-EGI/workflowEGI/src/main/java/it/infn/ct/DescribeVM.java

```diff
@@ -0,0 +1,341 @@
+                                if (address != null) System.out.println("occi.network.address = " + address);
+                                if (netid != null) System.out.println("org.opennebula.network.id = " + netid);
+                                if (netvlan != null) System.out.println("org.opennebula.network.vlan = " + netvlan);
+                                if (phydev != null) System.out.println("org.opennebula.network.phydev = " + phydev);
+                                if (bridge != null) System.out.println("org.opennebula.network.bridge = " + bridge);
+                        }
+                }
+
+		if (properties.getProperty("RESOURCE").equals("os_tpl"))
+		{
+			System.out.println("[ OS_TPL DESCRIPTION ]");
+			System.out.println("[[ " + properties.getProperty("OCCI_RESOURCE_ID") + " ]]");
+			String TERM = properties.getProperty("OCCI_RESOURCE_ID")
+				      .substring(properties.getProperty("OCCI_RESOURCE_ID").lastIndexOf("#") + 1);
+			List<Mixin> mixins = model.findRelatedMixins("os_tpl");
+		        for (Mixin mixin : mixins) {
+				if ( ((mixin.getTerm()).contains(TERM)) ) {
+					System.out.println("title: \t\t" + mixin.getTitle());
+			                System.out.println("term: \t\t" + mixin.getTerm());
+			                System.out.println("location: \t" + mixin.getLocation());
+				}
+			}
+		} 
+		
+		if (properties.getProperty("RESOURCE").equals("resource_tpl"))
+		{
+			// Getting the description of all the available template(s)
+			System.out.println("[ RESOURCE_TPL DESCRIPTION ]");
+			List<Mixin> mixins = model.findRelatedMixins(properties.getProperty("RESOURCE"));
+
+                        if (!mixins.isEmpty())
+       	        	        for (Mixin mixin : mixins) {
+					if (mixin.getTerm().equals(properties.getProperty("OCCI_RESOURCE_ID"))) {
+						System.out.println("[[ " + mixin.getLocation() + " ]]");
+						System.out.println("title: \t\t" + mixin.getTitle());
+                                                System.out.println("term: \t\t" + mixin.getTerm());
+						String locations = (mixin.getLocation()).toString();
+
+						String segments[] = locations.split("/");
+                       	                        System.out.println("location: \t" 
+						+ "/" + segments[segments.length - 1] + "/");
+					}
+				}
+		}
+
+	} catch (CommunicationException | RenderingException | AmbiguousIdentifierException ex) {
+		throw new RuntimeException(ex);
+	}
+
+    }
+
+    public static String[] describe (String vmID)
+    {
+    System.out.println("Debut de la description ici...");
+    System.out.println("vmID dans describe: " + vmID);
+	// [ Setting preferences here! ]
+	String AUTH = "x509"; 
+        String OCCI_ENDPOINT_HOST = "https://carach5.ics.muni.cz:11443"; // <= Change here!
+        //String OCCI_ENDPOINT_HOST = "http://stack-server.ct.infn.it:8787/occi1.1"; // <= Change here!
```

Comment:
Here and below a lot of commented code. If you are not planning to use this code, remove it